Lib/traceback.py

```diff
@@ -309,7 +321,7 @@ def walk_stack(f):
     current stack is used. Usually used with StackSummary.extract.
     """
     if f is None:
-        f = sys._getframe().f_back.f_back
+        f = sys._getframe().f_back.f_back.f_back.f_back
```

Comment:
Could you explain this? Why do we need to go extra 2 frames back

Reply:
Oh yes this is because now when you pass `walk_stack` to `StackSummary.extract` there are two additional methods being called: `_walk_tb_with_full_positions` and `extended_frame_gen`. Previous it was just `walk_tb` and `extract` so there were just two `f_back` calls.